### PR TITLE
Add debug code and asserts

### PR DESCRIPTION
**Pull request by charliegracie**
This code adds extra asserts and debugging to help track down an
occasional hang on Linux 390.

Issue: #2113

Signed-off-by: Charlie Gracie <charlie.gracie@gmail.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
genie-omr build all

**Comment by charliegracie** [comment]
genie-omr build all

File: fvtest/threadtest/abortTest.cpp [hunk 1]
```diff
@@ -204,15 +206,17 @@ TEST(ThreadAbortTest, Waiting)
 
 	omrthread_abort(t);
 
-	omrthread_monitor_wait(testdata.exitSync);
+	if (0 == testdata.aborted) {
+		omrthread_monitor_wait(testdata.exitSync);
```
Comment (0xdaryl) [inline fvtest/threadtest/abortTest.cpp]: Sorry, perhaps I haven't fully digested everything in this test, but what is this trying to achieve?

**Comment by 0xdaryl** [comment]
Also, a reminder to update the copyright header in this file.

File: fvtest/threadtest/abortTest.cpp (hunk 1 above)
Comment (charliegracie) [reply fvtest/threadtest/abortTest.cpp]: This ensures that if the other thread already did the notify (which it should not be able to) this thread will not wait indefinitely.

Comment (0xdaryl) [reply fvtest/threadtest/abortTest.cpp]: OK, thanks.

**Comment by 0xdaryl** [comment]
Just update the copyright header and this should be good to merge.

**Comment by 0xdaryl** [comment]
genie-omr build all

**Pull request merged**